client/src/components/expectations-list.tsx

```diff
@@ -0,0 +1,226 @@
+        >
+          <HintsList
+            classes={classes}
+            hints={expectation.hints}
+            updateHints={handleHintChange}
+          />
+        </Collapse>
+      </CardContent>
+    </Card>
+  );
+};
+
+const ExpectationsList = (props: {
+  classes: any;
+  expectations: LessonExpectation[];
+  updateExpectations: (val: LessonExpectation[]) => void;
+}) => {
+  const { classes, expectations, updateExpectations } = props;
+
+  function onDragEnd(result: DropResult) {
+    if (!result.destination) {
+      return;
+    }
+    const startIdx = result.source.index;
+    const endIdx = result.destination.index;
+    const [removed] = expectations.splice(startIdx, 1);
+    expectations.splice(endIdx, 0, removed);
+    updateExpectations([...expectations]);
+  }
+
+  function handleExpectationChange(val: string, idx: number): void {
+    expectations[idx].expectation = val;
+    updateExpectations([...expectations]);
+  }
+
+  function handleAddExpectation(): void {
+    const newItem = {
+      expectation: "Add a short ideal answer for an expectation, e.g. 'Red'",
+      hints: [
+        {
+          text:
+            "Add a hint to help for the expectation, e.g. 'One of them starts with R'",
+        },
+      ],
+    };
+    expectations.push(newItem);
+    updateExpectations([...expectations]);
+  }
+
+  function handleRemoveExpectation(idx: number): void {
+    expectations.splice(idx, 1);
+    updateExpectations([...expectations]);
+  }
+
+  function handleHintChange(val: Hint[], eIdx: number): void {
+    expectations[eIdx].hints = val;
+    updateExpectations([...expectations]);
+  }
+
+  return (
+    <Paper elevation={0} style={{ textAlign: "left" }}>
+      <Typography variant="body2" style={{ padding: 15 }}>
+        Expectations
+      </Typography>
+      <DragDropContext onDragEnd={onDragEnd}>
+        <Droppable droppableId="droppable">
+          {(provided, snapshot) => (
+            <List
+              {...provided.droppableProps}
+              ref={provided.innerRef}
+              className={
+                snapshot.isDraggingOver ? classes.listDragging : classes.list
+              }
+            >
+              {expectations.map((exp, i) => (
+                <Draggable
+                  key={`expectation-${i}`}
+                  draggableId={`expectation-${i}`}
+                  index={i}
+                >
+                  {(provided, snapshot) => (
+                    <ListItem
+                      ref={provided.innerRef}
+                      {...provided.draggableProps}
+                      {...provided.dragHandleProps}
+                    >
+                      <ExpectationCard
+                        classes={classes}
+                        expectation={exp}
+                        expIdx={i}
+                        canDelete={expectations.length > 1}
+                        handleExpectationChange={(val: string) => {
```

Comment:
same note: consider `useCallback` for functions passed to component props (but again maybe not necessary if no need for optimizing)